src/ZkFold/Base/Protocol/Protostar/Commit.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module ZkFold.Base.Protocol.Protostar.Commit (Commit (..), HomomorphicCommit (..), PedersonSetup (..)) where
+
+import           Data.Foldable                               (Foldable, toList)
+import           Prelude                                     (type (~), zipWith, ($), (<$>))
+import qualified Prelude                                     as P
+
+import           ZkFold.Base.Algebra.Basic.Class
+import           ZkFold.Base.Algebra.Basic.Number
+import           ZkFold.Base.Algebra.EllipticCurve.BLS12_381
+import           ZkFold.Base.Algebra.EllipticCurve.Class
+import           ZkFold.Base.Protocol.Protostar.Oracle
+
+-- | Commit to the object @a@ with commitment key @ck@ and results of type @f@
+--
+class Commit ck a f where
+    commit :: ck -> a -> f
+
+instance (RandomOracle ck x, RandomOracle a x, Ring x) => Commit ck a x where
+    commit ck a = oracle [oracle @ck @x ck, oracle @a @x a]
+
+-- | Homomorphic commitment scheme, i.e. (hcommit ck1 a1) * (hcommit ck2 a2) == hcommit (ck1 + ck2) (a1 + a2)
+--
+class HomomorphicCommit ck a c where
+    hcommit :: ck -> a -> c
+
+class PedersonSetup a where
+    pedersonGH :: (a, a)
+
+instance PedersonSetup (Point BLS12_381_G1) where
+    pedersonGH = (g, h)
+        where
+            -- Random points on BLS12_381_G1
+            -- The only requirement for them is so that nobody knows discrete logarithm of g base h
+            -- Keeping these numbers open seems safe as there is no known efficient algorithm to calculate discrete logarithm
+            -- TODO: Consider choosing these elements randomly each time instead of hardcoding them
```

Comment:
There is no known efficient algorithm, but as long as parameters are fixed, one can ran an inefficient algorithm long enough, so yeah, maybe we will choose them randomly soon